scripts/main/notifications.js

```diff
@@ -0,0 +1,30 @@
+let notifications = {
+	json: "",
+};
+
+notifications.update = function (params) {
+	if (params.email.length > 1) {
+		var regexp = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
```

Comment:
:astonished:

Ugh, it's past 11pm so my ability to parse crazy regexps is severely diminished. What is this _expected_ to pass? :smiley:

Reply:
haha, that will just validate for email format, so plain strings will not pass

Reply:
@d7415 Would you care to elaborate? What would you do?

Reply:
It's definitely a better pattern than most I've seen, but there will be valid email addresses that it misses. And the fact that it's so close almost makes it worse. Simplest example being that it special-cases ipv4 addresses, but makes no allowance for ipv6 (which are presumably valid). But also I think that you can use multiple quoted sections in the local part (which no-one is likely to do, but the pattern allows for one, which people are almost as unlikely to use...).

It's not a major issue, just minimal benefit. It doesn't prevent undeliverable emails (e.g user@example.com or abcde@fghi.jkl or abc@999.999.999.999), so why bother?